code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleExtension.java

```diff
@@ -0,0 +1,239 @@
+		lifecycleDataStore = getDataStore();
+		deviceInfoService = getDeviceInfoService();
+		lifecycleV1 = new LifecycleV1Extension(lifecycleDataStore, deviceInfoService, getApi());
+		lifecycleV2 = new LifecycleV2Extension(lifecycleDataStore, deviceInfoService, getApi());
+	}
+
+	/**
+	 * This constructor is intended for testing purposes.
+	 *
+	 * @param extensionApi {@code ExtensionApi} instance
+	 * @param namedCollection {@code NamedCollection} instance
+	 * @param deviceInfoService {@code DeviceInforming} instance
+	 */
+	@VisibleForTesting
+	protected LifecycleExtension(final ExtensionApi extensionApi,
+								 final NamedCollection namedCollection,
+								 final DeviceInforming deviceInfoService) {
+		super(extensionApi);
+		lifecycleDataStore = namedCollection;
+		this.deviceInfoService = deviceInfoService;
+		lifecycleV1 = new LifecycleV1Extension(lifecycleDataStore, deviceInfoService, getApi());
+		lifecycleV2 = new LifecycleV2Extension(lifecycleDataStore, deviceInfoService, getApi());
+	}
+
+	/**
+	 * This constructor is intended for testing purposes.
+	 *
+	 * @param extensionApi {@code ExtensionApi} instance
+	 * @param namedCollection {@code NamedCollection} instance
+	 * @param deviceInfoService {@code DeviceInforming} instance
+	 * @param lifecycleV1Extension {@code LifecycleV1Extension} instance
+	 * @param lifecycleV2Extension {@code LifecycleV2Extension} instance
+	 */
+	@VisibleForTesting
+	protected LifecycleExtension(final ExtensionApi extensionApi,
+								 final NamedCollection namedCollection,
+								 final DeviceInforming deviceInfoService,
+								 final LifecycleV1Extension lifecycleV1Extension,
+								 final LifecycleV2Extension lifecycleV2Extension) {
+		super(extensionApi);
+		lifecycleDataStore = namedCollection;
+		this.deviceInfoService = deviceInfoService;
+		this.lifecycleV1 = lifecycleV1Extension;
+		lifecycleV2 = lifecycleV2Extension;
+	}
+
+	@Override
+	protected String getName() {
+		return LifecycleConstants.EventDataKeys.Lifecycle.MODULE_NAME;
+	}
+
+	@Override
+	protected String getVersion() {
+		return Lifecycle.extensionVersion();
+	}
+
+	@Override
+	protected String getFriendlyName() {
+		return LifecycleConstants.FRIENDLY_NAME;
+	}
+
+	@Override
+	protected void onRegistered() {
+		getApi().registerEventListener(EventType.GENERIC_LIFECYCLE, EventSource.REQUEST_CONTENT, this::handleLifecycleRequestEvent);
+		getApi().registerEventListener(EventType.WILDCARD, EventSource.WILDCARD, this::updateLastKnownTimestamp);
+		lifecycleV1.processLifecycleExtensionRegistration();
+	}
+
+	@Override
+	public boolean readyForEvent(Event event) {
+		if (event.getType().equalsIgnoreCase(EventType.GENERIC_LIFECYCLE) && event.getSource().equalsIgnoreCase(EventSource.REQUEST_CONTENT)) {
+			SharedStateResult configurationSharedState = getApi().getSharedState(LifecycleConstants.EventDataKeys.Configuration.MODULE_NAME, event, false, SharedStateResolution.ANY);
+			if (configurationSharedState != null) {
```

Comment:
nit: simplify to (configurationSharedState != null) && configurationSharedState.status == SharedStateStatus.SET;